Zend/zend_types.h

```diff
@@ -670,9 +670,6 @@ static zend_always_inline uint32_t zval_gc_info(uint32_t gc_type_info) {
 #define Z_OBJPROP(zval)				Z_OBJ_HT((zval))->get_properties(&(zval))
 #define Z_OBJPROP_P(zval_p)			Z_OBJPROP(*(zval_p))
```

Comment:
I would add additional "for" argument to existing get_properties() handler, and change this into Z_OBJ_HT((zval))->get_properties(&(zval), ZEND_PROP_PURPOSE_DEFAULT)

Reply:
Actually, we might need additional ZEND_PROP_PURPOSE_FOREACH_READ and ZEND_PROP_PURPOSE_FOREACH_WRITE to avoid useless duplications in SPL ArrayObject.